File: backend/app/api/schemas/characters.py
from datetime import datetime

from pydantic import BaseModel


class CharacterListItem(BaseModel):
    id: int
    character_name: str
    corporation_name: str | None = None
    granted_scopes: list[str]
    sync_enabled: bool
    last_token_refresh: datetime | None = None
    last_successful_sync: datetime | None = None
    assets_sync_status: str
    orders_sync_status: str
    skills_sync_status: str
    structures_sync_status: str
    accessible_structure_count: int


class AccessibleStructureItem(BaseModel):
    structure_name: str
    structure_id: int
    system_name: str | None = None
    region_name: str | None = None
    access_verified_at: datetime
    tracking_enabled: bool
    polling_tier: str
    last_snapshot_at: datetime | None = None
    confidence_score: float


class CharacterDetail(BaseModel):
    id: int
    character_name: str
    corporation_name: str | None = None
    granted_scopes: list[str]
    sync_enabled: bool
    sync_toggles: dict[str, bool]
    structures: list[AccessibleStructureItem]
    skills: list[str]


class CharacterPatchRequest(BaseModel):
    sync_enabled: bool | None = None

File: frontend/src/types/characters.ts
export type CharacterListItem = {
  id: number;
  character_name: string;
  corporation_name: string | null;
  granted_scopes: string[];
  sync_enabled: boolean;
  last_token_refresh: string | null;
  last_successful_sync: string | null;
  assets_sync_status: string;
  orders_sync_status: string;
  skills_sync_status: string;
  structures_sync_status: string;
  accessible_structure_count: number;
};

export type AccessibleStructureItem = {
  structure_name: string;
  structure_id: number;
  system_name: string | null;
  region_name: string | null;
  access_verified_at: string;
  tracking_enabled: boolean;
  polling_tier: string;
  last_snapshot_at: string | null;
  confidence_score: number;
};

export type CharacterDetail = {
  id: number;
  character_name: string;
  corporation_name: string | null;
  granted_scopes: string[];
  sync_enabled: boolean;
  sync_toggles: Record<string, boolean>;
  structures: AccessibleStructureItem[];
  skills: string[];
};
